lib/chef/cookbook_version.rb

```diff
@@ -602,15 +594,23 @@ def <=>(o)
 
     private
 
+    def deprecated!(message)
+      if Chef::Config[:treat_deprecation_warnings_as_errors]
+        raise Exceptions::DeprecatedFeatureError, message
+      else
+        Chef::Log.warn(message)
```

Comment:
This could just be Chef::Log.deprecation(message) now or else use the mixin to deprecate methods?